### DISCUSSION TITLE
ObjectId cannot be used on equals step for Atlas Search

### DISCUSSION THREAD

**Discussion by ppascualv** [mongodb/mongoid#5543] [Q&A] [answered]
```
# frozen_string_literal: true

class SearchContact < BaseSearch

  def words_search
    Contact.collection.aggregate(aggregation_query).map { |c| c['_id'].to_s }
  end

  def aggregation_query
    [
      search_stage,
      { '$project': {
        _id: 1
      } }
    ].compact
  end

  def search_stage
    return unless words

    {
      '$search': {
        index: 'contact_search',
        compound: {
          filter: filter_stage,
          should: should_array
        }
      }
    }
  end

  def filter_stage
    [
      {
        equals: {
          path: 'account_id',
          value: {"$oid": current_user.account_id.to_s}
        }
      }
    ]
  end

  def should_array
    return [] unless words

    array = []
    %w[phone email first_name last_name second_last_name].each do |field|
      array << autocomplete(field)
    end

    array
  end

  def autocomplete(field)
    {
      autocomplete: {
        query: words,
        path: field
      }
    }
  end
end
```
The filter stage is not working returning that response:

[8:UnknownError]: PlanExecutor error during aggregation :: caused by :: Remote error from mongot :: caused by :: "compound.filter[0].equals.value" must be a boolean, objectId, number, or date (on develpment-main-api-shard-00-02.prt8v.mongodb.net:27017, modern retry, attempt 1)

How can I filter by ID? The same query works on SearchTester

**Comment by alexbevi** [comment, marked as answer]
Hi @ppascualv, the issue you appear to be facing is the pipeline being sent to the server that will be used by Atlas Search contains an invalid value type (`"compound.filter[0].equals.value" must be a boolean, objectId, number, or date`).

Instead of trying to convert `current_user.account_id` to EJSON, just let the driver serialize the value as needed.

Try the following and let us know if it works:
```ruby
  def filter_stage
    [
      {
        equals: {
          path: 'account_id',
          value: current_user.account_id
        }
      }
    ]
  end
```
